### DISCUSSION TITLE
exception in OnBeforeResourceLoad lead to not freeable

### DISCUSSION THREAD

**Discussion by trivalik** [cefsharp/CefSharp#3578] [Q&A]
We got by a programming issue a exception in OnBeforeResourceLoad in our derivation of ResourceRequestHandler. Then our programm tries to dispose anything. But in callstack it looks like the the Windows Forms control of CefSharp stuck. I could not identify the control since it hang in a unmanaged free function. Is this issue report worth?

**Comment by amaitland** [comment]
Make sure you don't propagate the exception, it's being called on an unmanaged thread (CEF IO thread).


https://bitbucket.org/chromiumembedded/cef/wiki/GeneralUsage.md#markdown-header-threads
https://github.com/cefsharp/CefSharp/wiki/General-Usage#threads

You can spawn a Task and continue processing asynchronously then execute the callback. 

> Is this issue report worth?

Without an actual example I cannot say definitively, it is unlikely there is anything actionable in terms of code changes to CefSharp.

**Reply by trivalik** [reply]
I just use it to add Bearer into `request`. If this is as expected then fine.
